### PR TITLE
Bluetooth: host: Add functions for periodic adv sync objects

### LINKED ISSUE #33612: Add support to get adv address of a per_adv_sync object and lookup per_adv_sync object from adv address
**Is your enhancement proposal related to a problem? Please describe.**
It will be useful for an application to lookup an advertiser address and see if there is already a periodic advertising sync object, or vice versa get the adv address of a periodic advertising sync object.

**Describe the solution you'd like**
Implement functions similar to `bt_conn_lookup_addr_le` and `bt_conn_get_dst` for periodic advertising sync objects. 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Thalley**
Adds two functions:
1) Lookup periodic adv sync objecy by address
2) Lookup address by per adv sync object

These can be useful for the application verify if there is already an
existing sync to an periodic advertiser.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/33612

### CODE REVIEW

File: include/bluetooth/bluetooth.h [hunk 1]
```diff
@@ -1290,6 +1290,22 @@ struct bt_le_per_adv_sync_param {
  */
 uint8_t bt_le_per_adv_sync_get_index(struct bt_le_per_adv_sync *per_adv_sync);
 
+/** @brief Get advertiser address address of a periodic adv sync object.
```
Comment (joerchan) [inline include/bluetooth/bluetooth.h:1293]: address address

That would be the address of the remote, right?
I'm wondering if in the future there would be other information needed and a more generic bt_le_per_adv_sync_get_info function would be more suited to retrieve this address, this is also something that we have
already for connections and extended advertiser.

Comment (Thalley) [reply include/bluetooth/bluetooth.h]: Fixed the address address. 

The get info might be a good alternative, as there are a few other fields that we might want to expose at a later time. Should these coexist, or should the get info replace this?

Comment (joerchan) [reply include/bluetooth/bluetooth.h]: I prefer replacing, it is similar for connections, addresses are included in the info.

File: include/bluetooth/bluetooth.h [hunk 2]
```diff
@@ -1290,6 +1290,41 @@ struct bt_le_per_adv_sync_param {
...
 
+/** @brief Advertising set info structure. */
+struct bt_le_per_adv_sync_info {
+	/** Periodic Advertiser Address */
+	bt_addr_le_t addr;
+
+	/** Advertiser SID */
+	uint8_t sid;
+
+	/** Periodic advertising interval (N * 1.25MS) */
```
Comment (joerchan) [inline include/bluetooth/bluetooth.h:1301]: Nitpick: 
```suggestion
	/** Periodic advertising interval (N * 1.25 ms) */
```

**Review by joerchan — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]
quick-ack

**Comment by Thalley** [comment]
@carlescufi Can this be merged soon please?

**Review by jhedberg — APPROVED** [review]

**Pull request merged**